### ISSUE TITLE
BucketAlreadyOwnedByYou exception on deploying first time into a new account

### ISSUE THREAD

**Issue by t-jones** [aws-cloudformation/rain#178]
Using rain to deploy the qnabot solution into a new account and encountered the following error:

```
$ rain deploy qnabot-on-aws-main.template QnABot --params 'Email=xxxxxxxxxxxxxx@amazon.com,ElasticSearchNodeCount=1,Encryption=UNENCRYPTED' -r us-west-2 -y -d
Rain needs to create an S3 bucket called 'rain-artifacts-xxxxxxxxxxxx-us-west-2'. Continue? (Y/n) y
unable to create artifact bucket 'rain-artifacts-xxxxxxxxxxxx-us-west-2': operation error S3: CreateBucket, https response error StatusCode: 409, RequestID: F2SAE3PRBP10FAK4, HostID: XtZK1ka1+Vok5KYf94oOJNsN5WC/jTKVEm18+EjbcrSMX0rMJ1S7h7BGzTTIddw8sdYXAkqKIZg=, BucketAlreadyOwnedByYou:
```

Deployment completed successfully when running again.  Is this an issue worth solving?  Maybe some sort of race condition?

**Comment by ericzbeard** [comment]
That's odd, you should only see that error if the bucket already exists. Doesn't seem likely that someone else would have run a rain deployment between the time it asked you if it could create the bucket and the time you continued. Here's the related code, it's not obvious how this is possible.

```go
	isBucketExists, err := BucketExists(bucketName)
	if err != nil {
		panic(fmt.Errorf("unable to confirm whether artifact bucket exists: %w", err))
	}

	if !isBucketExists {
		spinner.Pause()
		if !forceCreation && !console.Confirm(true, fmt.Sprintf("Rain needs to create an S3 bucket called '%s'. Continue?", bucketName)) {
			panic(errors.New("you may create the bucket manually and then re-run this operation"))
		}
		spinner.Resume()

		err := CreateBucket(bucketName)
		if err != nil {
			panic(fmt.Errorf("unable to create artifact bucket '%s': %w", bucketName, err))
		}
	}
```

**Comment by ericzbeard** [comment]
If you can reproduce this, I'd like to see the output when using the `--debug` flag.

**Comment by t-jones** [comment]
Saw this error again and captured debug output.  Running this against 10 new accounts it happened twice.

```
rain --debug deploy qnabot-on-aws-main.template QnABot --params 'Email=XXXXXXXXX@amazon.com,ElasticSearchNodeCount=1,Encryption=UNENCRYPTED' -r us-west-2 -y
DEBUG: Preparing template 'qnabot-on-aws-main.template'
DEBUG: Loading AWS config
DEBUG: Artifact bucket: rain-artifacts-XXXXXXXXXXXX-us-west-2
DEBUG: Packaging template: qnabot-on-aws-main.template

DEBUG: Matched: Resources/*|Type==AWS::Lambda::LayerVersion/Properties/Content

DEBUG: Matched: Resources/*|Type==AWS::Lambda::LayerVersion/Properties/Content

DEBUG: Matched: Resources/*|Type==AWS::Lambda::LayerVersion/Properties/Content

DEBUG: Matched: Resources/*|Type==AWS::Lambda::LayerVersion/Properties/Content

DEBUG: Matched: Resources/*|Type==AWS::Lambda::LayerVersion/Properties/Content

DEBUG: Matched: Resources/*|Type==AWS::Lambda::Function/Properties/Code

DEBUG: Matched: Resources/*|Type==AWS::Lambda::Function/Properties/Code

DEBUG: Matched: Resources/*|Type==AWS::Lambda::Function/Properties/Code

DEBUG: Matched: Resources/*|Type==AWS::Lambda::Function/Properties/Code

DEBUG: Matched: Resources/*|Type==AWS::Lambda::Function/Properties/Code

DEBUG: Matched: Resources/*|Type==AWS::Lambda::Function/Properties/Code

DEBUG: Matched: Resources/*|Type==AWS::Lambda::Function/Properties/Code

DEBUG: Matched: Resources/*|Type==AWS::Lambda::Function/Properties/Code

DEBUG: Matched: Resources/*|Type==AWS::Lambda::Function/Properties/Code

DEBUG: Matched: Resources/*|Type==AWS::Lambda::Function/Properties/Code

DEBUG: Matched: Resources/*|Type==AWS::Lambda::Function/Properties/Code

DEBUG: Matched: Resources/*|Type==AWS::Lambda::Function/Properties/Code

DEBUG: Matched: Resources/*|Type==AWS::Lambda::Function/Properties/Code

DEBUG: Matched: Resources/*|Type==AWS::Lambda::Function/Properties/Code

DEBUG: Matched: Resources/*|Type==AWS::Lambda::Function/Properties/Code

DEBUG: Matched: Resources/*|Type==AWS::Lambda::Function/Properties/Code

DEBUG: Matched: Resources/*|Type==AWS::Lambda::Function/Properties/Code

DEBUG: Matched: Resources/*|Type==AWS::Lambda::Function/Properties/Code

DEBUG: Matched: Resources/*|Type==AWS::Lambda::Function/Properties/Code

DEBUG: Matched: Resources/*|Type==AWS::Lambda::Function/Properties/Code

DEBUG: Matched: Resources/*|Type==AWS::Lambda::Function/Properties/Code

DEBUG: Matched: Resources/*|Type==AWS::Lambda::Function/Properties/Code

DEBUG: Matched: Resources/*|Type==AWS::Lambda::Function/Properties/Code

DEBUG: Matched: Resources/*|Type==AWS::Lambda::Function/Properties/Code

DEBUG: Matched: Resources/*|Type==AWS::CloudFormation::Stack/Properties/TemplateURL

DEBUG: Matched: Resources/*|Type==AWS::CloudFormation::Stack/Properties/TemplateURL

DEBUG: Matched: Resources/*|Type==AWS::CloudFormation::Stack/Properties/TemplateURL

DEBUG: Matched: Resources/*|Type==AWS::CloudFormation::Stack/Properties/TemplateURL

DEBUG: Matched: Resources/*|Type==AWS::CloudFormation::Stack/Properties/TemplateURL

DEBUG: Matched: Resources/*|Type==AWS::CloudFormation::Stack/Properties/TemplateURL

DEBUG: Checking current status of stack 'QnABot'
DEBUG: Handling parameters
DEBUG:   DefaultKendraIndexId:
DEBUG:   Encryption: UNENCRYPTED
DEBUG:   FulfillmentConcurrency: 0
DEBUG:   XraySetting: FALSE
DEBUG:   LLMSagemakerInitialInstanceCount: 1
DEBUG:   Username: Admin
DEBUG:   LexV2BotLocaleIds: en_US,es_US,fr_CA
DEBUG:   LexBotVersion: LexV2 Only
DEBUG:   SagemakerInitialInstanceCount: 1
DEBUG:   LLMSagemakerInstanceType: ml.g5.12xlarge
DEBUG:   Email: awstijon+team8@amazon.com
DEBUG:   ElasticSearchNodeCount: 1
DEBUG:   KibanaDashboardRetentionMinutes: 43200
DEBUG:   PublicOrPrivate: PRIVATE
DEBUG:   EmbeddingsApi: DISABLED
DEBUG:   EmbeddingsLambdaArn:
DEBUG:   LLMApi: DISABLED
DEBUG:   ElasticSearchEBSVolumeSize: 10
DEBUG:   InstallLexResponseBots: true
DEBUG:   EmbeddingsLambdaDimensions: 4096
DEBUG:   LLMLambdaArn:
DEBUG: Creating change set
DEBUG: Template is too large to deploy directly; uploading to S3.
DEBUG: Artifact bucket: rain-artifacts-XXXXXXXXXXXX-us-west-2
panic: error creating changeset: bucket does not exist: 'rain-artifacts-XXXXXXXXXXXX-us-west-2' [recovered]
	panic: error creating changeset: bucket does not exist: 'rain-artifacts-XXXXXXXXXXXX-us-west-2'

goroutine 1 [running]:
github.com/aws-cloudformation/rain/internal/cmd.execute.func1()
	github.com/aws-cloudformation/rain/internal/cmd/wrap.go:77 +0x178
panic({0x10215ede0, 0x140001428c0})
	runtime/panic.go:884 +0x1f4
github.com/aws-cloudformation/rain/internal/cmd/deploy.glob..func1(0x1025eade0?, {0x140000ab500, 0x2, 0x8?})
	github.com/aws-cloudformation/rain/internal/cmd/deploy/deploy.go:169 +0x1434
github.com/spf13/cobra.(*Command).execute(0x1025eade0, {0x140000ab480, 0x8, 0x8})
	github.com/spf13/cobra@v1.4.0/command.go:860 +0x4a8
github.com/spf13/cobra.(*Command).ExecuteC(0x1025ec1e0)
	github.com/spf13/cobra@v1.4.0/command.go:974 +0x338
github.com/spf13/cobra.(*Command).Execute(...)
	github.com/spf13/cobra@v1.4.0/command.go:902
github.com/aws-cloudformation/rain/internal/cmd.execute(0x60?)
	github.com/aws-cloudformation/rain/internal/cmd/wrap.go:86 +0x50
github.com/aws-cloudformation/rain/internal/cmd.Execute(0x102121d80?)
	github.com/aws-cloudformation/rain/internal/cmd/wrap.go:95 +0x1c
main.main()
	./main.go:23 +0x24
```
Thoughts?

**Comment by ericzbeard** [comment]
That's a different error message than the one originally reported. But I do think this must be a race condition. The `CreateBucket` API call must return success before the bucket is truly ready to use, so the `CreateChangeSet` call fails. I need to try to repro that with a simpler test, so I can figure out how to fix it reliably - maybe just a pause of a few seconds, but that seems like an overly simple solution that might not always work.

**Comment by ericzbeard** [comment]
Possibly the same issue as #213.

**Comment by ericzbeard** [comment]
@t-jones Let me know if this is still happening, I hope the fix for #213 works.
